File: src/jniglue/tests/org/OpenColorIO/LookTest.java

import junit.framework.TestCase;
import org.OpenColorIO.*;

public class LookTest extends TestCase {
    
    protected void setUp() {
    }
    
    protected void tearDown() {
    }
    
    public void test_interface() {
        Look lk = new Look().Create();
        lk.setName("coollook");
        assertEquals("coollook", lk.getName());
        lk.setProcessSpace("somespace");
        assertEquals("somespace", lk.getProcessSpace());
        lk.setDescription("this is a test");
        assertEquals("this is a test", lk.getDescription());
        ExponentTransform et = new ExponentTransform().Create();
        et.setValue(new float[]{0.1f, 0.2f, 0.3f, 0.4f});
        lk.setTransform(et);
        ExponentTransform oet = (ExponentTransform)lk.getTransform();
        float vals[] = new float[4];
        oet.getValue(vals);
        assertEquals(0.2f, vals[1], 1e-8);
        ExponentTransform iet = new ExponentTransform().Create();
        iet.setValue(new float[]{-0.1f, -0.2f, -0.3f, -0.4f});
        lk.setInverseTransform(iet);
        ExponentTransform ioet = (ExponentTransform)lk.getInverseTransform();
        float vals2[] = new float[4];
        ioet.getValue(vals2);
        assertEquals(-0.2f, vals2[1], 1e-8);
    }
    
}

File: src/python/tests/LookTest.py

import unittest, os, sys
sys.path.append(os.path.join(sys.argv[1], "src", "pyglue"))
import PyOpenColorIO as OCIO

class LookTest(unittest.TestCase):
    
    def test_interface(self):
        lk = OCIO.Look()
        lk.setName("coollook")
        self.assertEqual("coollook", lk.getName())
        lk.setProcessSpace("somespace")
        self.assertEqual("somespace", lk.getProcessSpace())
        lk.setDescription("this is a test")
        self.assertEqual("this is a test", lk.getDescription())
        et = OCIO.ExponentTransform()
        et.setValue([0.1, 0.2, 0.3, 0.4])
        lk.setTransform(et)
        oet = lk.getTransform()
        vals = oet.getValue()
        self.assertAlmostEqual(0.2, vals[1], delta=1e-8)
        iet = OCIO.ExponentTransform()
        iet.setValue([-0.1, -0.2, -0.3, -0.4])
        lk.setInverseTransform(iet)
        ioet = lk.getInverseTransform()
        vals2 = ioet.getValue()
        self.assertAlmostEqual(-0.2, vals2[1], delta=1e-8)
